### PR TITLE
Release/2.9.0

### PR DESCRIPTION
**Pull request by mhadam**

### CODE REVIEW

File: src/js/tracker.js [hunk 1]
```diff
@@ -1954,6 +2003,14 @@
 				businessUserId = userId;
 			},
 
+			/**
+			 * Alias for setUserId.
+			 *
+			 * @param string userId The business-defined user ID
+			 */
+
```
Comment (BenFradet) [inline src/js/tracker.js]: empty line

File: src/js/tracker.js [hunk 2]
```diff
@@ -1612,6 +1610,57 @@
 				return getPageViewId();
 			},
 
+			/**
+			 * Expires current session and starts a new session.
+			 *
```
Comment (BenFradet) [inline src/js/tracker.js]: empty line

File: core/lib/core.ts [hunk 3]
```diff
@@ -878,6 +878,84 @@ export function trackerCore(base64: boolean, callback?: (PayloadData) => void) {
...
+					description: description,
+				})
+			};
+
+			return trackSelfDescribingEvent({
+				schema: 'iglu:com.snowplowanalytics.snowplow/consent_granted/jsonschema/1-0-0',
+				data: removeEmptyProperties({
+					expiry: expiry,
+				})
+			}, context ? context.concat(documentJson) : [documentJson], tstamp);
```
Comment (BenFradet) [inline core/lib/core.ts]: Couldn't you add unit tests in: https://github.com/snowplow/snowplow-javascript-tracker/blob/master/core/tests/unit/core.js for both `trackConsentWithdrawn` and `trackConsentGranted`?

**Comment by mhadam** [comment]
@BenFradet let me know if there's any problems with the fixes I made

**Comment by mhadam** [comment]
I'm not sure if this is the issue causing it to stop - but the Safari suite failing seems to be caused by this:
https://github.com/theintern/intern/issues/657

**Comment by mhadam** [comment]
I've added documentation updates under the branch: [js-2.9.0](https://github.com/mhadam/snowplow.wiki-fork/tree/js-2.9.0)

File: src/js/forms.js [hunk 4]
```diff
@@ -58,6 +58,9 @@ object.getFormTrackingManager = function (core, trackerId, contextAdder) {
 	// Filter to determine which form fields should be tracked
 	var fieldFilter = function () { return true };
 
+	// Function applied to all elements
```
Comment (chuwy) [inline src/js/forms.js]: It would be nice to mention it can/will be overriden.

**Comment by mhadam** [comment]
@chuwy @BenFradet I separated out the core changes - they have their own issue branches now.

**Comment by alexanderdean** [comment]
Hey @BenFradet / @chuwy - is one of you available to show @mhadam the ropes in terms of getting an rc1 of his tracker live on the Snowplow website? Many thanks!

**Comment by chuwy** [comment]
Hey @BenFradet, I'm writing a short guide to @mhadam. Could you please later answer his questions (if he'll have any), as you're in a closer timezone?

**Comment by BenFradet** [comment]
@mhadam available to answer any qs today and tomorrow 👍

**Comment by alexanderdean** [comment]
Hey @chuwy - are you familiar with the release process for getting the JS Tracker rc into GTM? Is this something you can work with @mhadam on?

**Pull request merged**